server/metadata/tenant.go

```diff
@@ -192,10 +192,34 @@ func (m *TenantManager) CreateTenant(ctx context.Context, tx transaction.Tx, nam
 	return namespace, nil
 }
 
-func (m *TenantManager) GetNamespace(namespaceName string) Namespace {
+func (m *TenantManager) GetNamespace(ctx context.Context, namespaceName string, txMgr *transaction.Manager) Namespace {
 	m.RLock()
-	defer m.RUnlock()
 	if tenant, found := m.tenants[namespaceName]; found {
+		defer m.RUnlock()
+		return tenant.namespace
+	}
+	m.RUnlock()
+
+	// this will never create new namespace
+	// this is for reading namespaces from storage into cache
+	tx, err := txMgr.StartTx(ctx)
```

Comment:
take write lock just before this and do the same thing check again if it is already cached.

Reply:
done.